MyPlayground.playground/Contents.swift

```diff
@@ -0,0 +1,63 @@
+func arithmeticsFunction(array: [Int]) -> (arithmetic: Double, mediana: Double, moda: Set<Int>) {
+    if array.isEmpty { return (0, 0, Set<Int>()) }
+    
+    var sumOfNumbers = 0
+    var arithmeticNumber: Double
+    var medianaNumber: Double
+    var modaNumber = Set<Int>()
+    //var setOfNumbers = Set<Int>(array)
+    let arraySortedOfNumbers = array.sorted()
+    
+    // calculated mean
+    for numberOfArray in array {
+        sumOfNumbers += numberOfArray
+    }
+    arithmeticNumber = Double(sumOfNumbers)/Double(array.count)
+        
+    // calculated median
+    if arraySortedOfNumbers.count % 2 == 0 {
+       let prewiousNumberOfArray = arraySortedOfNumbers.count / 2 - 1
+       let nextNumberOfArray = arraySortedOfNumbers.count / 2
+       medianaNumber = (Double(arraySortedOfNumbers[prewiousNumberOfArray]) + Double(arraySortedOfNumbers[nextNumberOfArray])) / 2
+    } else {
+       let requiredNumberOfArray = Int(arraySortedOfNumbers.count / 2)
+       medianaNumber = Double(arraySortedOfNumbers[requiredNumberOfArray])
+    }
+    
+    // calculated moda
+    var numberOfItem = 0
+    var previousItem = 0  //optional
+    var entersOfNumber = [Int: Int]()
+    
+    for numberOfSortedArray in arraySortedOfNumbers {
```

Comment:
```suggestion
    for number in arraySortedOfNumbers {
```

I see that we're iterating the `arraySortedOfNumbers`.

Reply:
Fixed.